File: map-engine-opengl/src/jp/co/yahoo/map/codezine_sample/MapView.java
package jp.co.yahoo.map.codezine_sample;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.ArrayList;
import android.content.Context;
import android.widget.FrameLayout;

public class MapView extends FrameLayout{
	
	private GL20SurfaceView mSurfaceView = null;	//描画用ビュー
	private Context mContext = null;	//Context
	private float[] mCoordList;		//Float形式の座標を格納する配列
	
	//コンストラクタ
	public MapView(Context context,InputStream is) throws IOException {
		super(context);

		//地図データを読み込む
		mCoordList = loatData(is);

		//Contextを設定
		mContext = context;

		//描画のためのビューを初期化
		mSurfaceView = new GL20SurfaceView(mContext, this);

		//SurfaceViewを追加する
		this.addView(mSurfaceView);
	}

	//地図データを設定します。
	private float[] loatData(InputStream is) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(is));

		String str;
		ArrayList<String> list = new ArrayList<String>(); //行ごとのリスト
		//ファイルから１行ごとに読み込む
		while ((str = br.readLine()) != null) {
			list.add(str);
		}
		br.close();
		if(!(list.size()>0)) return null;

		//読み込んだファイルをFloat配列へ変換する
		float[] res = new float[list.size() * 3];  //x,y,zの順に格納されたfloat配列
		for(int i=0; i<list.size(); i++){
			str = list.get(i);
			String[] str_coord = str.split("\t");
			res[i * 3 + 0] = Float.valueOf(str_coord[0]);	//X座標
			res[i * 3 + 1] = Float.valueOf(str_coord[1]);	//Y座標
			res[i * 3 + 2] = 0.0f;							//Z座標
		}
		list.clear();

		return res;
	}

	//地図データを返します。
	public float[] getCoordinateList() {
		return mCoordList;
	}

	//地図データの初期化します。
	public void release() {
		mCoordList = null;
	}

}

File: MapEngineOpenGL/MapEngineOpenGL2.Droid/MapView.cs
using System;
using System.IO;

using Java.IO;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;

using Android.Util;
using System.Collections.Generic;

namespace MapEngineOpenGL
{
	public class MapView : FrameLayout, View.IOnTouchListener, IMapPresenter
	{
		private GestureDetector mGd;
		private GL20SurfaceView mSurfaceView = null;        //描画用ビュー
		private Context mContext = null;        //Context
		//private float[] mCoordList = null;                //Float形式の座標を格納する配列
		private MapControl mMc = null;        //地図操作用
		private MapController mSC = null; //共通処理
		DisplayMetrics mMetrics = null; //ディスプレイ情報

		public MapView (Context context, string filename) : base (context)
		{
			//ディスプレイ情報を取得する
			mMetrics = new DisplayMetrics();
			var wm = this.Context.GetSystemService (Context.WindowService).JavaCast<IWindowManager>();
			wm.DefaultDisplay.GetMetrics (mMetrics);

			//地図データを読み込む
			mSC = new MapController(this, filename);

			//Contextを設定
			mContext = context;

			//地図操作クラスを初期化
			mMc = new MapControl(this, mSC);
			mSC.SetEyePosition(140000.0f, 50000.0f, MapController.BASE_Z * mSC.scale, 0);

			//描画のためのビューを初期化
			mSurfaceView = new GL20SurfaceView(mContext, mSC);

			//SurfaceViewを追加する
			this.AddView(mSurfaceView);

			//ジェスチャー操作
			mGd = new GestureDetector(mContext, mMc);

			//タッチイベントを登録
			this.SetOnTouchListener(this);
		}

		/**
         * 地図データを返します。
         * @return
         */
		public float[] GetCoordinateList() {
			return mSC.CoordList;
		}

		/**
         * 地図操作クラスを返します。
         * @return
         */
		public MapControl GetMapControl() {
			return mMc;
		}

		/**
         * 地図データの初期化します。
         */
		public void Release() {
			mSC.Release ();
			mSC = null;
		}

		public void SetMatrix(GMatrix viewmatrix, GMatrix projectionmatrix)
		{
			this.GetSurfaceView ().SetMatrix (viewmatrix, projectionmatrix);
		}

		public bool OnTouch(View v, MotionEvent ev) {
			mGd.OnTouchEvent(ev);
			mMc.scaleGestureDetect.OnTouchEvent(ev);
			return true;
		}

		/**
         * SurfaceViewを取得
         * @return
         */
		public GL20SurfaceView GetSurfaceView() {
			return this.mSurfaceView;
		}
	}
}

